server/etcdserver/api/etcdhttp/health.go

```diff
@@ -46,33 +49,59 @@ type ServerHealth interface {
 // HandleHealth registers metrics and health handlers. it checks health by using v3 range request
 // and its corresponding timeout.
 func HandleHealth(lg *zap.Logger, mux *http.ServeMux, srv ServerHealth) {
-	mux.Handle(PathHealth, NewHealthHandler(lg, func(excludedAlarms AlarmSet, serializable bool) Health {
-		if h := checkAlarms(lg, srv, excludedAlarms); h.Health != "true" {
+	mux.Handle(PathHealth, NewHealthHandler(lg, func(excludedAlarms AlarmSet, serializable bool, endpoint string) Health {
+		if h := checkAlarms(lg, srv, excludedAlarms, endpoint); h.Health != "true" {
 			return h
 		}
 		if h := checkLeader(lg, srv, serializable); h.Health != "true" {
 			return h
 		}
 		return checkAPI(lg, srv, serializable)
-	}))
+	}, PathHealth))
+}
+
+// HandleLivez registers metrics and health handlers. it checks health by using v3 range request
+// and its corresponding timeout.
+func HandleLivez(lg *zap.Logger, mux *http.ServeMux, srv ServerHealth) {
+	mux.Handle(PathLivez, NewHealthHandler(lg, func(excludedAlarms AlarmSet, serializable bool, endpoint string) Health {
+		if h := checkAlarms(lg, srv, excludedAlarms, endpoint); h.Health != "true" {
+			return h
+		}
+		// TODO(logicalhan) should we require quorum for livez?
+		return checkAPI(lg, srv, serializable)
```

Comment:
```suggestion
		return checkAPI(lg, srv, true)
```

Reply:
`livez` doesn't require quorum

Reply:
You mean it shouldn't?

Reply:
I'm confused, do you mean 'false' here and 'true' on line 82?

Reply:
Also I have a todo here for a specific reason. I'm not even sure that this code is doing the right thing, since it's going through the etcdclient and therefore can hit any particular etcd member. So even if serializable is set to false and we're getting stale reads from a particular etcd member, we're not actually guaranteed to hit our own etcd member, which is the thing we need to be testing for.

Reply:
Right. `livez` just check local instance's health. We shouldn't restart the local instance if the cluster is NOT health, e.g. quorum isn't satisfied.

Reply:
Yes, but we shouldn't even be going through the etcd client, is my point, since it load balances.

Reply:
Ah, it's not going through the client if serializable is false. I see.